src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const totalItems: number = 42;
+  const values = [3, 5, 10, 20];
+  const [numItems, setNumItems] = useState(5);
+
+  const [currentPage, setCurrentPage] = useState(1);
+
+  let from = 1;
+  let to = numItems;
+
+  if (currentPage && currentPage !== 1) {
+    for (let page = 2; page <= currentPage; page++) {
+      from = to + 1;
+      to = numItems * page;
+
+      if (to > totalItems) {
+        to = totalItems;
+      }
+    }
+  }
+
+  const items = getNumbers(from, to).map(n => `Item ${n}`);
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${from} - ${to} of ${totalItems})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            className="form-control"
+            value={numItems}
+            onChange={event => {
+              setNumItems(Number(event.target.value));
+              setCurrentPage(1);
+            }}
```

Comment:
When changing the number of items per page, you should also check if the current page is still valid. If the new 'numItems' results in fewer pages, the 'currentPage' might be out of range. Add a check to reset 'currentPage' to the last available page if necessary.